compiler/compiler/src/compiler.rs

```diff
@@ -172,22 +172,64 @@ impl<'a, N: Network> Compiler<'a, N> {
         ))
     }
 
+    /// Run const propagation and loop unrolling until we hit a fixed point or find an error.
+    pub fn const_propagation_and_unroll_loop(&mut self, symbol_table: &mut SymbolTable) -> Result<()> {
+        loop {
+            let loop_unroll_output = self.loop_unrolling_pass(symbol_table)?;
```

Comment:
Just to confirm, is there any case where the alternate ordering of const prop then loop unroll would have made progress?